lib/main.go

```diff
@@ -200,6 +200,40 @@ func (obj *Main) Exit(err error) {
 	obj.exit <- err // trigger an exit!
 }
 
+// createPrefix handles logic to create (tmp) prefix directory.
+func (obj *Main) createPrefix(hostname string) (string, error) {
+	if obj.TmpPrefix {
+		// create temporary prefix
+		prefix, err := ioutil.TempDir("", fmt.Sprintf("%s-%s-", obj.Program, hostname))
+		if err != nil {
+			return prefix, fmt.Errorf("can't create tmp-prefix: %s", err)
+		}
+		log.Println("Main: Warning: Working prefix directory is temporary!")
+		return prefix, nil
+	}
+
+	// create/use default prefix
+	prefix := fmt.Sprintf("/var/lib/%s/", obj.Program) // default prefix
+	if p := obj.Prefix; p != nil {
+		prefix = *p
+	}
+	err := os.MkdirAll(prefix, 0770)
+	if err != nil {
+		if obj.AllowTmpPrefix {
+			// fallback to temporary prefix
+			prefix, err = ioutil.TempDir("", fmt.Sprintf("%s-%s-", obj.Program, hostname))
+			if err != nil {
+				return prefix, fmt.Errorf("can't create tmp-prefix: %s", err)
+			}
+			log.Println("Main: Warning: Working prefix directory is temporary!")
```

Comment:
Perhaps it's best to just extract all of these log messages and keep this to a single on in the main calling function there. Keep log messaging un-buried has some good benefits.